packages/frontend/src/components/connect-wallet/popovers/account/index.tsx

```diff
@@ -1,37 +1,118 @@
-import React, { useCallback, useMemo } from "react";
-import { Button, Popover } from "@carrot-kpi/ui";
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+import { Button, Popover, Typography } from "@carrot-kpi/ui";
 import { forwardRef } from "react";
-import { useConnect } from "wagmi";
+import { Address, useConnect, useEnsName, useNetwork } from "wagmi";
 import { ReadonlyConnector } from "../../../../connectors/readonly";
+import { shortenAddress } from "../../../../utils/address";
+import { Avatar } from "../../avatar";
+import { ReactComponent as Power } from "../../../../assets/power.svg";
+import { ReactComponent as External } from "../../../../assets/external.svg";
+import { ReactComponent as Copy } from "../../../../assets/copy.svg";
+import { ReactComponent as Tick } from "../../../../assets/tick.svg";
+import { useCopyToClipboard } from "react-use";
+import { cva } from "class-variance-authority";
+
+const copyAddressStyles = cva(["w-10", "h-10", "p-0"], {
+    variants: {
+        copiedToClipboard: {
+            true: [
+                "bg-green",
+                "hover:bg-green",
+                "text-black",
+                "hover:text-black",
+            ],
+        },
+    },
+});
 
 interface AccountPopoverProps {
+    address: string;
     open: boolean;
     onClose: () => void;
     anchor?: HTMLElement | null;
 }
 
 export const AccountPopover = forwardRef<HTMLDivElement, AccountPopoverProps>(
-    function AccountPopover({ open, anchor, onClose }, ref) {
+    function AccountPopover({ address, open, anchor, onClose }, ref) {
+        const { chain } = useNetwork();
+        const { data: ensName } = useEnsName({
+            address: address as Address,
+        });
         const { connectors, connect } = useConnect();
         const readonlyConnector = useMemo(() => {
             return connectors.find(
                 (connector) => connector instanceof ReadonlyConnector
             );
         }, [connectors]);
+        const blockExplorerHref = useMemo(() => {
+            if (!address || !chain || !chain.blockExplorers) return "";
+            return `${chain.blockExplorers.default.url}/address/${address}`;
```

Comment:
it would be great if a function return the correct url with no need for interpolation. Just pass it the address and current chain.

Could be something like this:

```suggestion
            return `blockEplorerUrl(chain, address);
```

Reply:
I don't think Wagmi supoprts it though, no?